notebook_helper/notebook_helper/importer/__init__.py

```diff
@@ -21,6 +21,61 @@ def run_cells(module, raise_on_error: bool = True):
     for cell in module.__cells__:
         cell.run(raise_on_error=raise_on_error)
 
+def run_cells_by_section(module, start_section, end_section = None, raise_on_error = True):
+    # Start_section is required as a string. It should include the section name (i.e. Section, Task, etc.)
+    # along with the section number (i.e. 1, 2, 3) or letter (i.e. A, B, C). Please make sure the captialization
+    # matches the markdown in the notebook
+
+    # Creating initial empty global variables
+    start, end = 0, None
+    prefix = None
+    start_found = False
+
+    # Loop through the cells of the notebook
+    for i, cell in enumerate(module.__cells__):
+        # Only scan markdown cells for headings
+        if getattr(cell, 'cell_type', None) != "markdown":
+            continue
+
+        # Scan all lines in the markdown cell
+        lines = cell.source.splitlines() if cell.source else []
+
+        for line in lines:
+            if not start_found:
+                # Detect start_section at the beginning of a line
+                start_pattern = re.match(rf"(#+) {start_section}", line)
+                if start_pattern:
+                    start_found = True
+                    start = i
+                    prefix = start_pattern.group(1) + " "  # '#' characters + space
+            # Once start is found, look for end_section
+            else:
+                # If an end_section parameter is given
+                if end_section:
+                    if re.match(rf"{prefix}{end_section}", line):
```

Comment:
I don't think we should require `end_section` to be at the same header level as `start_section`. So instead of using `prefix`, please just use the `#+` pattern.

Reply:
Yes, exactly